### PR TITLE
fix(kit): sync store products end to end

### PR DESCRIPTION
**Pull request by hyochan**
## Summary
- Fix App Store Connect and Play Console product sync for create/update/delete flows.
- Activate Play one-time purchase options, preserve consumable types on pull, and surface delete counts.
- Add Petgu-based IAPKit E2E skill and clarify iOS-only metadata guidance in the products UI.

## Verification
- [x] bun run typecheck
- [x] bun run test -- convex/products
- [x] bun run lint:eslint
- [x] pre-commit kit gate: install, lint, prettier, full kit tests, smoke server
- [x] Petgu App Store Connect / Play Console push-pull-delete sync verified with temporary SKUs

## Notes
- No preview recording: this is primarily sync/backend behavior. The visible UI change is a conditional iOS-only note on the Products form, verified locally.

## Summary by CodeRabbit

* **New Features**
  * Product sync now deletes upstream store products for rows marked **Removed** during **push/both** syncs, and reports **deleted** counts in job results.
  * Sync UI messaging and banners now clearly differentiate between standard syncs and **purge-local** runs.
* **Bug Fixes**
  * **Removed** kit products are now preserved during **pull** syncs and won’t be unintentionally restored.
  * Android one-time product sync now better preserves kit-authored product types and improves pricing handling for edge cases.
* **Tests**
  * Added coverage for ASC JWT issuer behavior and kit upsert state defaults.

### CODE REVIEW

File: packages/kit/convex/products/play.ts [hunk 1]
```diff
@@ -860,11 +919,255 @@ async function performAndroidSync(
...
   };
 }
 
+function googleErrorStatus(error: unknown): number | undefined {
+  if (!error || typeof error !== "object") return undefined;
+  const candidate = error as { code?: unknown; status?: unknown };
+  if (typeof candidate.code === "number") return candidate.code;
+  if (typeof candidate.status === "number") return candidate.status;
+  return undefined;
+}
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/play.ts:943]: Fixed in 9818763b. googleErrorStatus now handles numeric/string code values and gaxios response.status.

[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/play.ts:943]: Fixed in 9818763b. googleErrorStatus now handles numeric/string code values and gaxios response.status.

File: packages/kit/convex/products/play.ts [hunk 2]
```diff
@@ -860,11 +919,255 @@ async function performAndroidSync(
...
+  }
+
+  if (options.allowCreate) {
+    await insertLegacyAndroidOneTimeProduct(androidpublisher, args);
+    await activateAndroidOneTimePurchaseOption(auth, args);
+    return;
+  }
+  await patchLegacyAndroidOneTimeProduct(androidpublisher, args);
+  await activateAndroidOneTimePurchaseOption(auth, args);
+}
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/play.ts:979]: Fixed in 9818763b. One-time products now use the raw modern endpoint first and activate purchase options only after that path succeeds; legacy fallback no longer calls purchase-option activation.

File: packages/kit/convex/products/play.ts [hunk 3]
```diff
@@ -860,11 +919,255 @@ async function performAndroidSync(
...
+    return;
+  }
+
+  if (options.allowCreate) {
+    await insertLegacyAndroidOneTimeProduct(androidpublisher, args);
+    await activateAndroidOneTimePurchaseOption(auth, args);
+    return;
+  }
+  await patchLegacyAndroidOneTimeProduct(androidpublisher, args);
+  await activateAndroidOneTimePurchaseOption(auth, args);
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/play.ts]: Fixed in 9818763b. One-time products now use the raw modern endpoint first and activate purchase options only after that path succeeds; legacy fallback no longer calls purchase-option activation.

File: packages/kit/convex/products/asc.ts [hunk 4]
```diff
@@ -72,16 +72,19 @@ async function resolveAscCredentials(
     ? (project.iosAscIssuerId ?? project.iosAppStoreIssuerId)
     : project.iosAppStoreIssuerId;
   const keyId = useAsc ? project.iosAscKeyId : project.iosAppStoreKeyId;
-  if (!issuerId || !keyId) {
+  if (!keyId) {
+    const missing = [
+      ...(!keyId ? [useAsc ? "iosAscKeyId" : "iosAppStoreKeyId"] : []),
+    ];
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/asc.ts]: Fixed in 9818763b. Simplified the key-id missing-field array.

File: packages/kit/convex/products/play.ts [hunk 5]
```diff
@@ -385,15 +388,24 @@ async function performAndroidSync(
...
                 })
               : undefined;
+            const existingType = await ctx.runQuery(
+              internal.products.sync.getExistingProductType,
+              {
+                projectId: project._id,
+                platform: "Android",
+                productId: product.productId,
+              },
+            );
```
[the remark this answers is not in the document]
Comment (hyochan) [reply packages/kit/convex/products/play.ts]: Fixed in 9818763b. Android pull now loads existing product types once into an in-memory map instead of querying per product.

**Comment by hyochan** [comment]
Addressed the latest CodeRabbit follow-ups in 87f445b9.\n\nChanges:\n- setSubPriceSchedule now routes through /v1/subscriptionPrices.\n- modern Play one-time delete no longer masks non-404 errors.\n- deleteRemovedProductRow re-checks kit-authored removal eligibility.\n- IAPKit Petgu E2E skill paths now use environment placeholders.
